src/handlers/psp.ts

```diff
@@ -165,45 +176,63 @@ const initializeState = (config: Config): State => {
 
 const executeApiCalls = async (state: State): Promise<State> => {
   const { config, baseLogOptions, groupedSubscriptions } = state;
-
-  // TODO: promise.all? 🤔
-  let apiValuesBySubscriptionId: { [subscriptionId: string]: ethers.BigNumber } = {};
-  for (const { subscriptions, template, endpoint } of groupedSubscriptions) {
-    const templateLogOptions: node.LogOptions = {
-      ...baseLogOptions,
-      additional: {
-        templateId: template.id,
-      },
-    };
+  const apiValuePromises = groupedSubscriptions.map(({ subscriptions, template, endpoint }) => {
     const apiCallParameters = abi.decode(template.templateParameters);
-    const logsData = await go(
+
+    return go(
       () =>
-        callApi({
-          oises: config.ois,
-          apiCredentials: config.apiCredentials,
-          apiCallParameters,
-          oisTitle: endpoint.oisTitle,
+        callApi(config, {
+          id: template.id,
```

Comment:
> I'm starting to really dislike the node.CallApiPayload and how tied it is to RRPrequests.
> This should work but I find it weird that we need to pass the airnodeAddress here since it's only required for '[regular](https://github.com/api3dao/airnode/blob/master/packages/airnode-node/src/api/index.ts#L36)' request type.

> Now that I see this I'm remembering the verification steps in node.callApi()
1. I think it doesn't make sense to send templateId as requestId
2. verification of requestId won't make sense
3. verification of the sponsor wallet seems that will be done twice if we call node.callApi()

I think the above points are both related to the types required by node.callApi() so we could simplify `callApiOptions` if we were to make these fields optional in the **airnode-node** `AggregatedApiCall` type.

@acenolaza @amarthadan Any objections for me to make these (type) changes to Airnode? Or please let me know if you have other ideas.

Reply:
AFAIK no verification will be done for `http-gateway` type we want to use here.

I agree that the current design of `callApi` is not really prepared for a wide variety of requests and that should be changed. But for now I would suggest keeping up with the current `callApi` interface even though some values might not make sense. I'll create a task for future refactoring of `callApi`. Do you agree?

Reply:
You are right @amarthadan about the verification being skipped when using http-gateway. I missed the checks that return null on each verify* function. i. e. https://github.com/api3dao/airnode/blob/master/packages/airnode-node/src/api/index.ts#L130

I'm on board with the idea to make it work and then we make it pretty but I would prefer if we keep the [callApi](https://github.com/api3dao/airkeeper/blob/main/src/api/call-api.ts#L9)() interface clean and simple so rrp.ts and psp.ts would call this function the same way and with just the minimum required args. Then this required by the node.callApi() can be hard-coded/added inside this function and if we then change the node side no changes would be required in rrp.ts and psp.ts...just in call-api.ts. This is only a suggestion and can be disregarded at will.

Reply:
Totally agreed, let's do it that way :muscle:

Reply:
> I'm on board with the idea to make it work and then we make it pretty but I would prefer if we keep the [callApi](https://github.com/api3dao/airkeeper/blob/main/src/api/call-api.ts#L9)() interface clean and simple so rrp.ts and psp.ts would call this function the same way and with just the minimum required args.

Sounds great :smile: